File: cpp/trojkat.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main(int argc, char **argv)
{
    int a, b, c;
    float p = 0;
    a = b = c = 0;
    cout << "Podaj 3 boki trójkąta." << endl;
    cout << "Bok a:";
    cin >> a;
    cout << "Bok b:";
    cin >> b;
    cout << "Bok c:";
    cin >> c; 
    p = (a+b+c)/2;
    
    if (a+b>c && a+c>b && b+c>a)
    {
    cout << "Obwód trójkąta jest równy:" << a+b+c << endl;
    cout << "Pole trójkąta jest równe:" << sqrt(p*(p-a)*(p-b)*(p-c)) << endl;
    }
    
    return 0;
}

File: kurs/trojkat.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import math


def main(args):
    a, b, c = eval(input('Podaj boki trójkąta oddzielone przecinkami: '))
    print('Podano boki: {}, {}, {}'.format(a, b, c))
    trojkat = False
    if a + b > c:
        if a + c > b:
            if b + c > a:
                trojkat = True
    if trojkat:
        print('Da się zbudować trójkąt!')
        p = (a + b + c) / 2
        pole = math.sqrt(p * (p - a) * (p - b) * (p - c))
        print('Pole trójkąta o bokach {}, {}, {}'.format(
            a, b, c), ': ', '{:.4f}'.format(pole))
        if (a**2 + b**2 == c**2) or \
            (a**2 + c**2 == b**2) or \
                (c**2 + b**2 == a**2):
            print('Trójkąt jest prostokątny!')
        else:
            print('Trójkąt nie jest prostokątny!')
    else:
        print('Nie da sie!')

    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
